lib/hammer_cli_foreman/attribute.rb

```diff
@@ -0,0 +1,303 @@
+        params['compute_attribute']['vm_attrs']['volumes_attributes'] = HammerCLIForeman::Attribute.attribute_hash(option_volume_list) unless option_volume_list.empty?
+        params
+      end
+
+      success_message _('Compute profile attributes are set.')
+      failure_message _('Could not set the compute profile attributes')
+      build_options
+    end
+
+    class Update < HammerCLIForeman::UpdateCommand
+      desc _('Update compute profile values.')
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+
+      extend_help do |h|
+        ::HammerCLIForeman::ComputeResources.extend_help(h, :all)
+      end
+
+      validate_options do
+        any(:option_compute_profile_id, :option_compute_profile_name ).required
+        any(:option_compute_resource_id, :option_compute_resource_name).required
+      end
+
+
+      def request_params
+
+        params = HammerCLIForeman::Attribute.get_params(options)
+        raise ArgumentError, "Compute profile value to update does not exist yet; it needs to be created first" if !params['compute_attribute'].key?('id')
+        params['id'] =  params['compute_attribute']['id']
+        vm_attrs = params['compute_attribute']['vm_attrs']
+        original_volumes = vm_attrs['volumes_attributes'] || {}
+        original_interfaces = vm_attrs['interfaces_attributes'] || {}
+
+        if options['option_compute_attributes']
+          vm_attrs = options['option_compute_attributes']
+          vm_attrs['volumes_attributes'] ||= original_volumes
+          vm_attrs['interfaces_attributes'] ||= original_interfaces
+        end
+        vm_attrs['interfaces_attributes'] = HammerCLIForeman::Attribute.attribute_hash(options['option_interface_list']) unless options['option_interface_list'].empty?
+        vm_attrs['volumes_attributes'] = HammerCLIForeman::Attribute.attribute_hash(options['option_volume_list']) unless options['option_volume_list'].empty?
+        params['compute_attribute']['vm_attrs'] = vm_attrs
+        params
+
+      end
+      success_message _('Compute profile attributes updated.')
+      failure_message _('Could not update the compute profile attributes')
+
+      build_options :without => :id
+    end
+
+    # Using the Update command because adding a new interface is done by modifying existing compute_attribute
+    class AddInterface < HammerCLIForeman::UpdateCommand
+      command_name 'add-interface'
```

Comment:
somehow when doing `hammer compute-profile values add-interface -h`, provider specific options don't get populated:
```
 hammer compute-profile values add-interface -h
Usage:
    hammer compute-profile values add-interface [OPTIONS]

Options:
 --compute-profile COMPUTE_PROFILE_NAME    Compute profile name
 --compute-profile-id COMPUTE_PROFILE_ID    
 --compute-resource COMPUTE_RESOURCE_NAME  Compute resource name
 --compute-resource-id COMPUTE_RESOURCE_ID  
 --interface SET_VALUES                    Interface parameters, should be comma separated list of values
                                           Comma-separated list of key=value
 --location LOCATION_NAME                  Location name
 --location-id LOCATION_ID                  
 --location-title LOCATION_TITLE           Location title
 --organization ORGANIZATION_NAME          Organization name
 --organization-id ORGANIZATION_ID         Organization ID
 --organization-title ORGANIZATION_TITLE   Organization title
 -h, --help                                Print help

Provider specific options:
  ec2:


  gce:


  libvirt:


  openstack:


  ovirt:


  rackspace:


  vmware:
```
This happens only for add-interface other values subcommand have provider specific help as expected

Reply:
done